src/Session.cpp

```diff
@@ -92,14 +93,32 @@ Session::Session(const Session &s)
     clean();
 }
 
+void Session::deepCopyUsers(const std::unordered_map<std::string, User*>& newUsers)
+{
+    for(auto u: newUsers)
+    {
+        *(userMap[u.first()]) = *(u.second());
+    }
+}
+
+template<typename T>
+void Session::deepCopyPointerVector(const std::vector<T*>& newV, std::vector<T*>& ourV)
+{
+    for(auto x: newV)
+    {
+        //create a new T object using T's copy constructor
+        ourV.push_back(new T(*x));
```

Comment:
same, you should do 
for (auto x : *this)
{
    delete x
}
and then ...

Reply:
the function runs in Session, which has no definition for an iterator. It should only copy from the other

Reply:
I agree to both, you are right. just please write in function documentation that it only copies new map/vector and not existing one